lib/util/transformer/length-based-splitter.js

```diff
@@ -1,11 +1,19 @@
-/* eslint-disable promise/prefer-await-to-callbacks */
 import Stream from 'stream';
 import log from '../../logger';
 
 class LengthBasedSplitter extends Stream.Transform {
-
-  constructor (readableStream, littleEndian, maxFrameLength, lengthFieldOffset, lengthFieldLength, lengthAdjustment) {
+  constructor (opts) {
```

Comment:
opts = {}

Reply:
I don't think this should be defaulted. The person who writes the calling code should know when they mess up and do not initialize. Defaulting just hides where that happens.